cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ValidationUtils.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.datapipeline.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfigurer;
+import io.cdap.cdap.etl.api.Engine;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.batch.BatchPipelineSpec;
+import io.cdap.cdap.etl.batch.BatchPipelineSpecGenerator;
+import io.cdap.cdap.etl.common.DefaultPipelineConfigurer;
+import io.cdap.cdap.etl.common.DefaultStageConfigurer;
+import io.cdap.cdap.etl.proto.v2.ETLBatchConfig;
+import io.cdap.cdap.etl.proto.v2.ETLPlugin;
+import io.cdap.cdap.etl.proto.v2.ETLStage;
+import io.cdap.cdap.etl.proto.v2.spec.StageSpec;
+import io.cdap.cdap.etl.proto.v2.validation.StageSchema;
+import io.cdap.cdap.etl.proto.v2.validation.StageValidationRequest;
+import io.cdap.cdap.etl.proto.v2.validation.StageValidationResponse;
+import io.cdap.cdap.etl.spec.PipelineSpecGenerator;
+import io.cdap.cdap.etl.validation.ValidatingConfigurer;
+import io.cdap.cdap.internal.io.SchemaTypeAdapter;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Utility functions for common pipeline validation code
+ */
+public class ValidationUtils {
```

Comment:
final and private constructor for util class.

Reply:
Done